ShellCheck/Analytics.hs

```diff
@@ -3670,6 +3670,33 @@ checkNonportableSignals _ = checkUnqualifiedCommand "trap" (const f)
         return $ err id 2173
             "SIGKILL/SIGSTOP can not be trapped."
 
+prop_checkMkdir0 = verify checkMkdir "mkdir -p -m 0755 dir"
+prop_checkMkdir1 = verify checkMkdir "mkdir -pm 0755 dir"
+prop_checkMkdir2 = verify checkMkdir "mkdir -vpm 0755 dir"
+prop_checkMkdir3 = verify checkMkdir "mkdir -pm 0755 -v dir"
+prop_checkMkdir4 = verifyNot checkMkdir "mkdir -p dir"
+prop_checkMkdir5 = verifyNot checkMkdir "mkdir -m 0755 dir"
+prop_checkMkdir6 = verifyNot checkMkdir "mkdir dir"
+checkMkdir _ = checkCommand "mkdir" (const $ f (Nothing,Nothing)) where
+    f _ [] = return ()
+    f ((Just mIdent),(Just pIdent)) _ = do warn (getId mIdent) 2174 $ "-m will be ignored for parent directories"
+    f (dashMIdent,dashPIdent) (a:r) =
```

Comment:
You can simplify this with https://github.com/koalaman/shellcheck/blob/master/ShellCheck/ASTLib.hs#L110 

E.g. something like:

    potentially $ do
      dashP <- find ((== "p") . snd) flags
      dashM <- find ((== "m") . snd) flags
      return $ warn (getId $ fst dashM) 2174 "-m will be ignored"
    where flags = getAllFlags cmd in